library/std/src/net/udp.rs

```diff
@@ -90,6 +90,12 @@ impl UdpSocket {
     /// ];
     /// let socket = UdpSocket::bind(&addrs[..]).expect("couldn't bind to address");
     /// ```
+    ///
+    /// Note that `bind` will limit which addresses you can `send_to` and
+    /// `connect` to. Binding to localhost as in the example above,
+    /// will prevent you from sending to another device in your local network
+    /// (e.g. `192.168.2.100`). In order to limit yourself the least, you can
+    /// `bind` to [`Ipv4Addr::UNSPECIFIED`] or [`Ipv6Addr::UNSPECIFIED`].
```

Comment:
This is just a feature of networks in general, and the higher-level principle probably should be explained: when you bind, you create a way to address _you_ on the network, and thus create a view of the network... both to and from you.

So, binding to the loopback address gives you the loopback view of the network and now no one can address you except via the loopback address, which they can only reach if they are already there, because that's pretty much the definition of loopback anything: it doesn't go anywhere.

Reply:
I'll address this and try to improve the writing.